tests/utils/test_shell_util.py

```diff
@@ -360,6 +364,134 @@ def test_run_pipe_should_return_a_bytes_object_when_encode_output_is_false(self)
 
         self.assertTrue(isinstance(output, bytes), "The return value should be a bytes object. Got: '{0}'".format(type(output)))
 
+    # R0912: Too many branches (13/12) (too-many-branches) -- Disabled: Branches are sequential
+    def test_run_command_run_pipe_run_get_output_should_notify_when_commands_start_and_complete(self):  # pylint:disable=R0912
+        # The children processes run this script, which creates a file (this signals that process has started) and then sleeps for a long time
+        child_script = os.path.join(self.tmp_dir, "should_notify_when_commands_start_and_complete.py")
+        AgentTestCase.create_script(child_script, """
+import os
+import sys
+import time
+
+with open(sys.argv[1], "w") as signal_file:
+    signal_file.write("{0} {1}".format(os.getpid(), os.getppid()))
+time.sleep(60)
+""")
+
+        threads = []
+
+        try:
+            child_processes = []
+            parent_processes = []
+
+            started_commands = []
+            completed_commands = []
+
+            # W0108: Lambda may not be necessary (unnecessary-lambda) - The use of lambda is appropriate
+            shellutil.set_on_command_started_callback(lambda pid: started_commands.append(pid))  # pylint:disable=W0108
+            shellutil.set_on_command_completed_callback(lambda pid: completed_commands.append(pid))  # pylint:disable=W0108
```

Comment:
My understanding of `W0108` is that, instead of `lambda pid: started_commands.append(pid)`, you can just do `started_commands.append`.

See, e.g., [this StackOverflow post](https://stackoverflow.com/a/56446811).

Reply:
Yes, either way is correct functionally. For me, it is clearer to say in the code that I am passing a callable that takes a pid and appends it to the array, it is more explicit that just passing append.

Reply:
I personally agree about the readability. I think it would be beneficial to modify the comment to point out that benefit, but I don't think it's really that big of a deal. Your discretion!

Reply:
I would advocate for disabling all the style rules in Pylint; I find that most of the time they just get in the way.

Reply:
+1 to that, if not disable all then atleast disable most (like too many branches, not enough instances, etc etc)